public/app/plugins/datasource/cloudwatch/datasource.js

```diff
@@ -191,7 +185,20 @@ function (angular, _) {
           });
         }
 
-        return this.getDimensionValues(region, namespace, metricName, dimensions);
+        return this.getDimensionValues(region, namespace, metricName, dimensions).then(function(result) {
+          return _.map(result, function(dimensions) {
+            var values = _.chain(dimensions)
+            .sortBy(function(dimension) {
+              return dimension.Name;
+            })
+            .map(function(dimension) {
+              return dimension.Name + '=' + dimension.Value;
```

Comment:
@torkelo 
One of the idea is,
Storing "dim_key1=dim_value1,dim_key2=dim_value2..." value to "$dimensions", and expand string like "$dimensions[0].name" to each dimension name/value at this function.
https://github.com/grafana/grafana/blob/32f9f8fcceba3bcfd47bc398e524178019ed2faa/public/app/plugins/datasource/cloudwatch/datasource.js#L279-L286

Or, add input field to query editor, the field can take "dim_key1=dim_value1,dim_key2=dim_value2...".
And, parse this format string, then set(overwrite) dimension keys/values at this line.
https://github.com/grafana/grafana/blob/32f9f8fcceba3bcfd47bc398e524178019ed2faa/public/app/plugins/datasource/cloudwatch/datasource.js#L37